tests/e2e/magic_functions/functions/c_read.cpp

```diff
@@ -0,0 +1,135 @@
+// Copyright 2022 Memgraph Ltd.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+// License, and you may not use this file except in compliance with the Business Source License.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+#include <stdexcept>
+
+#include "mg_procedure.h"
+
+static void return_function_argument(struct mgp_list *args, mgp_func_context *ctx, mgp_func_result *result,
+                                     struct mgp_memory *memory) {
+  mgp_value *value{};
+  auto err_code = mgp_list_at(args, 0, &value);
+  if (err_code != MGP_ERROR_NO_ERROR) {
+    mgp_func_result_set_error(result, "Failed to fetch list!", memory);
+    return;
+  }
+
+  err_code = mgp_func_result_set_value(result, value, memory);
+  if (err_code != MGP_ERROR_NO_ERROR) {
+    mgp_func_result_set_error(result, "Failed to construct return value!", memory);
+    return;
+  }
+}
+
+double get_element_from_arg(struct mgp_list *args, int index) {
+  mgp_value *value{nullptr};
+  if (mgp_list_at(args, index, &value) != MGP_ERROR_NO_ERROR) {
+    throw std::runtime_error("Error while argument fetching.");
+  }
+
+  double result;
+  int is_int;
+  mgp_value_is_int(value, &is_int);
+
+  if (is_int) {
+    int64_t result_int;
+    mgp_value_get_int(value, &result_int);
+    result = static_cast<double>(result_int);
+  } else {
+    mgp_value_get_double(value, &result);
+  }
+  return result;
+}
+
+static void add_two_numbers(struct mgp_list *args, mgp_func_context *ctx, mgp_func_result *result,
+                            struct mgp_memory *memory) {
+  double first = 0;
+  double second = 0;
+  try {
+    first = get_element_from_arg(args, 0);
+    second = get_element_from_arg(args, 1);
+  } catch (...) {
+    mgp_func_result_set_error(result, "Unable to fetch the result!", memory);
+    return;
+  }
+
+  mgp_value *value{nullptr};
+  auto summation = first + second;
+  mgp_value_make_double(summation, memory, &value);
+
+  auto err_code = mgp_func_result_set_value(result, value, memory);
+  if (err_code != MGP_ERROR_NO_ERROR) {
+    mgp_func_result_set_error(result, "Failed to construct return value!", memory);
+  }
+}
+
+static void return_null(struct mgp_list *args, mgp_func_context *ctx, mgp_func_result *result,
+                        struct mgp_memory *memory) {
+  mgp_value *value{};
+  mgp_value_make_null(memory, &value);
```

Comment:
free this at the end! Consider, our or `OnScopeExit` util function! (same change for all the occurrences of mgp_value_make_*** functions around this file.